eventgrid/data-plane/src/main/java/com/microsoft/azure/eventgrid/customization/EventGridSubscriber.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.eventgrid.customization;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.eventgrid.models.EventGridEvent;
+import com.microsoft.azure.management.apigeneration.Beta;
+import com.microsoft.azure.serializer.AzureJacksonAdapter;
+import com.microsoft.rest.protocol.SerializerAdapter;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The type that can be used to de-serialize events.
+ */
+@Beta
+public class EventGridSubscriber {
+    /**
+     * The default adapter for to be used for de-serializing the events
+     */
+    private final AzureJacksonAdapter defaultSerializerAdapter;
+    /**
+     * The map containing user defined mapping of eventType to Java model type
+     */
+    private Map<String, Type> eventTypeToEventDataMapping;
+
+    @Beta
+    public EventGridSubscriber() {
+        this.defaultSerializerAdapter = new AzureJacksonAdapter();
```

Comment:
If the Data property contains polymorphic types, will this handle that as well, or do we need some special logic for it?  In the C# case, we do something like:

           // Note: If any of the events have polymorphic data, add converters for them here.
            // This enables the polymorphic deserialization for the event data.
            // For example, MediaJobCompletedEventData's JobOutput type is polymorphic 
            // based on the @odata.type property in the data.

            // Example usage: jsonSerializer.Converters.Add(new PolymorphicDeserializeJsonConverter<JobOutput>("@odata.type"));

Reply:
Looking at the tests, I do see you are supporting this, which is great! However, it wasn't quite clear to me whether this is handled automatically, or whether like in C# we need to express something like the above where we add the PolymorphicDeserializeJsonConverter explicitly.

Reply:
De-serialization of polymorphic types are handled using Jackson annotations. See this models https://github.com/Azure/azure-sdk-for-java/blob/e6771250bb50c5b38a301afed06afb88c6737a08/eventgrid/data-plane/src/test/java/com/microsoft/azure/eventgrid/customization/models/ShippingInfo.java#L15

Reply:
I see, thanks. For 1st party / system event data that contains polymorphic fields, will we have to annotate in the same way (on top of the default autorest generated class)?

Reply:
For system events - if discriminator is defined in swagger for corresponding event models then auto rest will generate these annotations, no need to modify the generated models.

For custom events - User has to annotate their model as shown in the tests.

Reply:
sounds good, thanks!